### PR TITLE
Switch to using a single registry value for all images.

### LINKED ISSUE #225: OLCNE - Kubernetes module requires --container-registry
When creating the kubernetes module, the option "--container-registry" is required. Since it is not provided by default it fails with

`master1: FATAL[24/04/20 18:29:25] Error while validating flags. required flags: "container-registry"`

### PR DESCRIPTION
**Pull request by Djelibeybi**
This should resolve any weirdness with adding new modules manually as well.

This should fix #225, #226 and #227.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

### CODE REVIEW

**Comment by rafabene** [comment]
It fixed #225 and #227. Trying to install Istio now to validate the PR

**Comment by rafabene** [comment]
However now I'm facing the following issue:

> master1:     olcnectl module install --environment-name olcne-env --name olcne-cluster
    master1: Returned a non-zero code: 1
    master1: Last output lines:
    master1: To troubleshoot, list all containers using your preferred container runtimes CLI, e.g. docker.
    master1: Here is one example how you may list all Kubernetes containers running in docker:
    master1: 	- 'docker ps -a | grep kube | grep -v pause'
    master1: 	Once you have found the failing container, you can inspect its logs with:
    master1: 	- 'docker logs CONTAINERID'

**Comment by Djelibeybi** [comment]
It's probably timing out trying to download images. I recommend using an Oracle Container Registry mirror closer to your location by leveraging the `vagrant-env` plugin and creating a `.env.local` file that sets the `REGISTRY_OLCNE` value to a mirror of OCR.

Check https://docs.cloud.oracle.com/en-us/iaas/Content/General/Concepts/regions.htm for a list of all available regions. There is an OCR mirror in every region and it has various aliases based on the Region Name, Region Identifier and Region Key. 

For example, `container-registry-saopaulo.oracle.com`, `container-registry-sa-saopaulo-1.oracle.com` or `container-registry-gru.oracle.com` all point to the same mirror in Brazil.

**Comment by rafabene** [comment]
However the docker reference in the binary should be replaced to use podman.

**Comment by Djelibeybi** [comment]
It's an example. The line above states that you should use your preferred container runtime, which in our case is CRI-O/`crictl` anyway. This is standard Kubernetes and is not changed by either OLCNE or the Vagrant build.

**Comment by rafabene** [comment]
Everything worked when I commented out --nginx-image. Please, check that using a registry mirror created by registry-image-helper.sh . Plus olcne-helm and olcne-istio images are not mirrored by registry-image-helper.sh

**Comment by Djelibeybi** [comment]
That's a good point. We'll need to open up a bug to get `registry-image-helper.sh` updated with the new OLCNE images. Thanks!

**Comment by AmedeeBulle** [comment]
For the mirroring issue, as workaround, this is what I use to mirror images to my local registry:

```shell
#!/bin/bash

registry_from="container-registry.oracle.com/olcne"
registry_to="my.local.mirror/olcne"

images="\
  nginx:1.12.2 \
  citadel:1.4.6 \
  galley:1.4.6 \
  istio_kubectl:1.4.6 \
  mixer:1.4.6 \
  pilot:1.4.6 \
  proxyv2:1.4.6 \
  sidecar_injector:1.4.6 \
  prometheus:v2.13.1 \
  grafana:v6.4.4 \
  "

for i in ${images}; do
  podman pull ${registry_from}/${i}
  podman tag ${registry_from}/${i} ${registry_to}/${i}
  podman push  ${registry_to}/${i}
done
```

File: OLCNE/scripts/provision.sh [hunk 1]
```diff
@@ -530,8 +522,8 @@ deploy_kubernetes() {
     echo_do olcnectl module create \
       --environment-name "${OLCNE_ENV_NAME}" \
       --module kubernetes --name "${OLCNE_CLUSTER_NAME}" \
-      ${REGISTRY_K8S:+--container-registry $REGISTRY_K8S} \
-      ${REGISTRY_OLCNE:+--nginx-image $REGISTRY_OLCNE/$NGINX_IMAGE} \
+      --container-registry "${REGISTRY_OLCNE}" \
+      --nginx-image "${REGISTRY_OLCNE}/${NGINX_IMAGE}" \
```
[the remark this answers is not in the document]
Comment (AmedeeBulle) [reply OLCNE/scripts/provision.sh:526]: No it does not -- see my code snippet above, it includes the nginx image.

**Comment by rafabene** [comment]
I'm still having the following error when I use a mirror registry created by **registry-image-helper.sh ** It only works when I remove *--nginx-image*

>master1: Returned a non-zero code: 1
    master1: Last output lines:
    master1: To troubleshoot, list all containers using your preferred container runtimes CLI, e.g. docker.
    master1: Here is one example how you may list all Kubernetes containers running in docker:
    master1: 	- 'docker ps -a | grep kube | grep -v pause'
    master1: 	Once you have found the failing container, you can inspect its logs with:
    master1: 	- 'docker logs CONTAINERID'
    master1: See /var/tmp/cmd_1cvy5.log for details

**Comment by Djelibeybi** [comment]
> I'm still having the following error when I use a mirror registry created by **registry-image-helper.sh ** It only works when I remove _--nginx-image_

This is a problem with `registry-image-helper.sh` not the Vagrant box. I've already logged a bug for this internally. It won't work until that's fixed. In the interim, please use the alternative script provided by @AmedeeBulle earlier in this pull request (which really need to be merged, so I'm going to merge it anyway).

**Comment by AmedeeBulle** [comment]
Verified against default registry, OCI mirror and local registry mirror for both single and multi master.

**Pull request merged**